icaruscode/Decode/DecoderTools/TriggerDecoderV3_tool.cc

```diff
@@ -823,17 +839,52 @@ namespace daq
      * * east wall:  `00<C3P2><C3P1><C3P0>00<C2P2><C2P1><C2P0>`
      * * west wall:  `00<C1P2><C1P1><C1P0>00<C0P2><C0P1><C0P0>`
      * Therefore, the two 32-bit half-words need to be swapped
-     * This holds for both cryostats, and both walls.  
+     * This holds for both cryostats, and both walls.
+     * (note that the `00` bits may actually contain information from adders)
      */
 
```

Comment:
Could you explain how the adder information comes in here from the hardware? Is it solely part of the LVDS information and passed on through the data fragment information?

Reply:
The adder output is discriminated, and the discriminator output for each of the 12 adder channels is shoved into the two 64-bit FPGA connectors by the same custom hardware box that also processes the LVDS signals from the readout boards. These boxes have a 24 bit ribbon input designed for the readout boards, and additional 6 (?) single inputs. The box takes care of sorting out all this into the 64 bit output to the FPGA. The FPGA code uses the 24 bits of each of the four half-connectors to build the majority, and the other bits to build the adder trigger logic. The FPGA has hard-coded all the locations of the PMT pairs and of the adders, so that it knows which to combine. On a global trigger, the FPGA reads the 2 64-bit connectors again, it unpacks the bit information and packs it back into two 64-bit words, which are the ones written in the trigger "string". I do not remember the extent of the repackaging, but I recall it's not extensive (i.e. it's close to a copy).
Repeat that for each cryostat.
At least the hardware side of this should be documented in some of Donatella Torretta presentations.